python/lsst/obs/base/defineVisits.py

```diff
@@ -655,12 +742,121 @@ def group(self, exposures: List[DimensionRecord]) -> Iterable[VisitDefinitionDat
                 e.group_id == visitId for e in exposuresInGroup
             ), "Grouping by exposure.group_name does not yield consistent group IDs"
             yield VisitDefinitionData(
-                instrument=instrument, id=visitId, name=visitName, exposures=exposuresInGroup
+                instrument=instrument,
+                id=visitId,
+                name=visitName,
+                exposures=exposuresInGroup,
+                visit_system_mask=2**self.config.visitSystemId,
             )
 
-    def getVisitSystem(self) -> Tuple[int, str]:
+    def getVisitSystems(self) -> List[Tuple[int, str]]:
+        # Docstring inherited from GroupExposuresTask.
+        return [(self.config.visitSystemId, self.config.visitSystemName)]
+
+
+class _GroupExposuresByCounterAndExposuresConfig(GroupExposuresConfig):
+    visitSystemId = Field(
+        doc="Integer ID of the visit_system implemented by this grouping algorithm.",
+        dtype=int,
+        default=5,  # Not appliccable. One-to-one ID OR by-group-counter ID.
```

Comment:
If Instrument subclasses are going to need to specify a default visit system, we can't have the value for the visit system being a configurable item that any user can change. I think we are going to have to have a public class that defines all the visit systems so that a particular ID can be specified by name by an instrument.

Reply:
I have been thinking that it would be ideal to find a way to put the actual allowed values into the dimensions configuration YAML file, by introducing some way for a dimension's "table" to be backed by a very small number of constant values.  I didn't want to hold up the schema change by pushing that, though, because it would require some Python-side changes, too.  Declaring the values constant for all instruments via code in obs_base sounds like a good alternative.  Could even be an actual Python enum instead of the strings we've got now.

Reply:
I've switched to an enum and no longer use the config values I also removed visit_system_mask and everything looks a lot cleaner now.